acceptance/tests/options/custom_facts.rb

```diff
@@ -42,4 +45,32 @@
   on(agent, facter("--custom-dir '#{custom_dir}' custom_fact")) do
     assert_equal("testvalue", stdout.chomp, "Custom fact output does not match expected output")
   end
+
+  on(agent, "rm -f '#{custom_fact}'")
+
+  step "Agent #{agent}: ensure custom facts in $FACTERLIB resolve"
+  facterlib_dir = agent.tmpdir('arbitrary_dir')
+  custom_fact = File.join(facterlib_dir, 'custom_fact.rb')
+  create_remote_file(agent, custom_fact, content)
+  on(agent, "chmod +x '#{custom_fact}'")
+
+  on(agent, facter('custom_fact', :environment => { 'FACTERLIB' => facterlib_dir })) do
+    assert_equal("testvalue", stdout.chomp, "Custom fact output does not match expected output")
+  end
+
+  on(agent, "rm -rf '#{facterlib_dir}'")
+
+  step "Agent #{agent}: ensure custom facts in $LOAD_PATH resolve"
+  on(agent, "#{ruby_command(agent)} -e 'puts $LOAD_PATH[0]'")
+  load_path_facter_dir = File.join(stdout.chomp, 'facter')
+  on(agent, "mkdir -p \"#{load_path_facter_dir}\"")
+  custom_fact = File.join(load_path_facter_dir, 'custom_fact.rb')
+  create_remote_file(agent, custom_fact, content)
+  on(agent, "chmod +x '#{custom_fact}'")
+
+  on(agent, facter("custom_fact")) do
+    assert_equal("testvalue", stdout.chomp, "Custom fact output does not match expected output")
+  end
+
+  on(agent, "rm -rf '#{load_path_facter_dir}'")
```

Comment:
Are there any circumstances where LOAD_PATH[0]/facter would already exist, and be something we don't want to delete?